test/unittest/tests/test_acquire.cpp

```diff
@@ -0,0 +1,39 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+
+#include "wt_internal.h"
+
+#define TEST_TYPE(type)                               \
+    TEST_CASE("Test acquiring a " #type, "[acquire]") \
+    {                                                 \
+        type a = 5;                                   \
+        type *ap = &a;                                \
+        type a_result;                                \
+        WT_ACQUIRE_READ(a_result, *ap);               \
+        REQUIRE(a == a_result);                       \
+                                                      \
+        a = 10;                                       \
+        WT_ACQUIRE_READ_WITH_BARRIER(a_result, *ap);  \
+        REQUIRE(a == a_result);                       \
+        REQUIRE(true);                                \
+    }
+
+/*
+ * We can't test that the specific instructions we want are being output, but having written this
+ * test we found that clang failed to compile for sizes that weren't uint64_t without some
```

Comment:
Detecting the instruction emitted should be possible with a clever shell script and objdump, but I'm happy to stick with the current implementation.

I don't see anything in evergreen.yml that guarantees we run the unit tests with clang. Can you update the `! Ubuntu 20.04 Minimal` variant with `CMAKE_TOOLCHAIN_FILE: -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_stable_clang.cmake` and a comment explaining why we need this?

Reply:
4131291c4 and 570d1cdb7